zebra/zebra_nhg.c

```diff
@@ -2190,13 +2224,27 @@ static unsigned nexthop_active_check(struct route_node *rn,
 		break;
 	}
 
+skip_check:
+
 	if (!CHECK_FLAG(nexthop->flags, NEXTHOP_FLAG_ACTIVE)) {
 		if (IS_ZEBRA_DEBUG_RIB_DETAILED)
 			zlog_debug("        %s: Unable to find active nexthop",
 				   __func__);
 		return 0;
 	}
 
+	/* Capture recursive nexthop mtu.
+	 * TODO -- the code used to just reset the re's value to zero
+	 * for each nexthop, and then jam any resolving route's mtu value in,
+	 * whether or not that was zero, or lt/gt any existing value? The
+	 * way this is used appears to be as a floor value, so let's try
+	 * using it that way here.
+	 */
+	if (mtu > 0) {
+		if (re->nexthop_mtu == 0 || re->nexthop_mtu > mtu)
```

Comment:
could be collapsed into the surrounding if

Reply:
there are already some pretty "compressed" blocks that are sort of hard to parse, so I made this a little more open or expanded. are you saying "could" but it's ok to leave it be, or ... "could" meaning "change it ?